generator/generator.py

```diff
@@ -1,31 +1,101 @@
-#import numpy
+import numpy as np
 import sys
-import csv
 import random
 
-def generator(shards, deltaT, N, mean, std):
-    tmp_list = []
+def generator(num_of_shards, delta_t, size_of_vectors, mean, std):
     vectors = []
+    tasks_in_delta_t = []
+    time_marks_in_delta_t = []
+    requests = []
 
-    for x in range(shards):
-        for y in range(N):
-            tmp_list.append(random.randint(0, 10))
-        vectors.append(tmp_list)
-        tmp_list = []
+    vectors = create_vectors(int(num_of_shards), int(size_of_vectors), float(mean), float(std))
 
+    save_vectors(vectors)
+            
+    tasks_in_delta_t = count_positive_tasks(int(size_of_vectors), vectors)
 
+    time_marks_in_delta_t = generate_time_stamps(int(size_of_vectors), float(delta_t), tasks_in_delta_t)
+    
+    requests = create_requests(int(size_of_vectors), tasks_in_delta_t, time_marks_in_delta_t, vectors)
+    
+    add_indexes_to_requests(requests)
+
+    save_requests(requests)
+
+def create_vectors(num_of_shards, size_of_vectors, mean, std):
+    vectors = []
+    for shard in range(num_of_shards): 
+        loads = []    
+        loads = np.random.normal(mean, std, size_of_vectors).tolist()
+        loads = [round(load, 3) for load in loads]
+        vectors.append(loads)
+    return vectors
+
+def save_vectors(vectors):
     vectors_file = open("./generator/vectors.csv","w")
-    for x in vectors:
-        vector_string = str(vectors.index(x)+1) +","+ ','.join(map(str, x)) + "\n"
-        vectors_file.write(vector_string)
+    for vector in vectors:
+       vector_string = str(vectors.index(vector)+1) +","+ ','.join(map(str, vector)) + "\n"
+       vectors_file.write(vector_string)
+    vectors_file.close()
+
+def count_positive_tasks(size_of_vectors, vectors):
+    tasks_in_delta_t = []
+    for column in range(size_of_vectors):
+        num_of_positive_loads = 0
+        for vector in vectors:
+            if(vector[column]>0):
+                num_of_positive_loads = num_of_positive_loads + 1
+        tasks_in_delta_t.append(num_of_positive_loads)
```

Comment:
misleading is that `tasks_in_delta_t` is an array, it should be an integer, all vectors should be same size and we don't want any to have any negative element in any vector - then we should throw some exception, and break the program